pico-slack.js

```diff
@@ -35,10 +35,11 @@ const processTeamInfo = (teamInfo)=>{
 	}
 };
 
-const handleEvent = (rawData, flags)=>{
+const handleEvent = async (rawData, flags)=>{
 	try {
 		const evt = JSON.parse(rawData);
 		if(evt.error) return Slack.error(evt);
+		if(evt.type === 'goodbye') await reconnect();
```

Comment:
Oh, I love that you can just insert this here and then continue with processing and emission of the `goodbye` event! So clean.

Reply:
Yeah originally I halted processing, but then Scott mentioned something about emitting an event so consumers can still close the socket if they really want to, and it turned out to be very easy :)